src/Message/PurchaseRequest.php

```diff
@@ -69,60 +115,93 @@ public function getData()
-            'payment'       => $this->getPaymentMethod(),
-            'purchaseid'    => $this->getTransactionId(),
-            'amount'        => $this->getAmountInteger(),
-            'issuerid'      => $this->getIssuer(),
-            'entrancecode'  => $this->getEntranceCode(),
-            'description'   => $this->getDescription(),
-            'including'     => $this->getIncluding(),
-            'days'          => $this->getDays(),
-            'returnurl'     => $this->getReturnUrl(),
-            'cancelurl'     => $this->getCancelUrl(),
-            'notifyurl'     => $this->getNotifyUrl(),
-            'sha1'          => $this->generateSignature(),
-            'testmode'      => $this->getTestMode(),
+            'shopid' => $this->getShopId(),
+            'merchantid' => $this->getMerchantId(),
+            'merchantkey' => $this->getMerchantKey(),
+            'payment' => $this->getPaymentMethod(),
+            'purchaseid' => $this->getTransactionId(),
+            'amount' => $this->getAmountInteger(),
+            'issuerid' => $this->getIssuer(),
+            'entrancecode' => $this->getEntranceCode(),
+            'description' => $this->getDescription(),
+            'including' => $this->getIncluding(),
+            'days' => $this->getDays(),
+            'returnurl' => $this->getReturnUrl(),
+            'cancelurl' => $this->getCancelUrl(),
+            'notifyurl' => $this->getNotifyUrl(),
+            'sha1' => $this->generateSignature(),
+            'testmode' => $this->getTestMode(),
         );
 
         /** @var \Omnipay\Common\CreditCard $card */
         $card = $this->getCard();
         if ($card) {
-            if ($this->getPaymentMethod() == 'overboeking' || $this->getPaymentMethod() == 'ecare') {
-                $data['billing_mail']       = $card->getEmail();
-                $data['billing_firstname']  = $card->getBillingFirstName();
-                $data['billing_lastname']   = $card->getBillingLastName();
+            if ($this->getPaymentMethod() == 'overboeking' || $this->getPaymentMethod() == 'klarna') {
+                $data['billing_mail'] = $card->getEmail();
+                $data['billing_firstname'] = $card->getBillingFirstName();
+                $data['billing_lastname'] = $card->getBillingLastName();
             }
 
-            if ($this->getPaymentMethod() == 'ecare') {
-                $data['billing_company']    = $card->getBillingCompany();
-                $data['billing_address1']   = $card->getBillingAddress1();
-                $data['billing_address2']   = $card->getBillingAddress2();
-                $data['billing_zip']        = $card->getBillingPostcode();
-                $data['billing_city']       = $card->getBillingCity();
-                $data['billing_country']    = $card->getBillingCountry();
-                $data['billing_phone']      = $card->getBillingPhone();
+            if ($this->getPaymentMethod() == 'klarna') {
```

Comment:
According to the technical support member I spoke with on the phone yesterday, all support for ecare has been removed. They reused the term for the non technical customers on their website, however the technical implementation is now longer supported.